internal/bminventory/inventory.go

```diff
@@ -5191,6 +5206,15 @@ func (b *bareMetalInventory) UpdateInfraEnvInternal(ctx context.Context, params
 	if pullSecretUpdated {
 		params.InfraEnvUpdateParams.PullSecret = pullSecretBackup
 	}
+	return params
```

Comment:
I feel like this doesn't need to return anything. We're not passing a pointer to the update params and the only thing that's changing is the pull secret which also isn't a pointer.

So this whole exercise of resetting is probably unnecessary at this point.

Now that it's a separate method it would probably be better to just dup the update params, log them and forget having to track if the pull secret was changed or not.